server/src/main/kotlin/com/kamelia/hedera/rest/auth/SessionManager.kt

```diff
@@ -77,9 +84,18 @@ object SessionManager {
             )
         }
         val tokenData = TokenData.from(user)
-        val session = Session(userState, tokenData)
 
-        sessions[tokenData.accessToken] = session
+        if (sessionId == null) {
+            val oldTokens = refreshTokens[refreshToken] ?: throw ExpiredOrInvalidTokenException()
+            val session = sessions[oldTokens.accessToken] ?: throw ExpiredOrInvalidTokenException()
+            sessions[tokenData.accessToken] = session.copy(user = userState, tokenData = tokenData)
+            sessions.remove(oldTokens.accessToken)
```

Comment:
`remove` already returns the removed element, so instead of getting above and then removing here, just remove